preprocessor/plugin_functions.F90

```diff
@@ -0,0 +1,136 @@
+!    Copyright (C) 2007 Imperial College London and others.
+!    
+!    Please see the AUTHORS file in the main source directory for a full list
+!    of copyright holders.
+!
+!    Prof. C Pain
+!    Applied Modelling and Computation Group
+!    Department of Earth Science and Engineering
+!    Imperial College London
+!
+!    amcgsoftware@imperial.ac.uk
+!    
+!    This library is free software; you can redistribute it and/or
+!    modify it under the terms of the GNU Lesser General Public
+!    License as published by the Free Software Foundation,
+!    version 2.1 of the License.
+!
+!    This library is distributed in the hope that it will be useful,
+!    but WITHOUT ANY WARRANTY; without even the implied warranty of
+!    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+!    Lesser General Public License for more details.
+!
+!    You should have received a copy of the GNU Lesser General Public
+!    License along with this library; if not, write to the Free Software
+!    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+!    USA
+
+#include "fdebug.h"
+module plugin_functions
+
+  use fields
+  use global_parameters, only: OPTION_PATH_LEN, PYTHON_FUNC_LEN, is_active_process
+
+
+  implicit none
+
+
+
+
+private
+
+public :: set_from_plugin_function
+
+
+
+contains
+
+
+  subroutine set_from_plugin_function(field, plugin_name,function_name, position, time)
+
+    use iso_c_binding
+    use dl_fortran
+    
+    implicit none
+
+    type( scalar_field ), intent( inout ) :: field
+    type( vector_field ), intent( in), target     :: position
+    real, intent (in )                    :: time
+
+    character (len=*),  intent( in )      :: plugin_name,function_name
+
+
+    character (len=1024), pointer :: fstring
+    integer :: slen
+    type(C_PTR) :: handle=C_NULL_PTR
+    type(C_FUNPTR) :: funptr=C_NULL_FUNPTR
+
+    integer(C_INT) :: status
+    type( vector_field ), target :: lposition
+    type( vector_field ), pointer :: fposition
+
+    integer :: stat
+
+    abstract interface
```

Comment:
I might be wrong, but I don't think this needs to be an abstract interface. In general we don't (yet) depend on the OO features of F2003 in Fluidity, and I'm not sure they have been fully implemented in all compilers we use.